lib/registry/src/publish.rs

```diff
@@ -112,11 +179,19 @@ pub fn try_chunked_uploading(
             package.version
         )
     })?;
+    Ok(url)
+}
 
-    let signed_url = url.url;
-    let url = url::Url::parse(&signed_url).unwrap();
+fn upload_package(
+    signed_url: &str,
+    archive_path: &PathBuf,
+    archived_data_size: u64,
+    timeout: Duration,
+) -> Result<(), anyhow::Error> {
+    let url = url::Url::parse(signed_url).unwrap();
```

Comment:
I know this shouldn't ever fail, but we're returning `anyhow::Error` anyway so may as well do proper error handling.

```suggestion
    let url = url::Url::parse(signed_url).context("Invalid signed URL")?;
```

Reply:
Added the context, closing this thread.